### PR TITLE
Understanding Predictive Mean Matching and Regression Modeling in R

### LINKED ISSUE #6787: Handling missing data: understanding predictive mean matching and regression modelling in R
### Topic Suggestion
Handling missing data: understanding predictive mean matching and regression modelling in R

#### Proposed title of article
Understanding predictive mean matching and regression modelling in R

#### Proposed article introduction
Any real-world datasets have missing data, which causes problems for both modeling and analysis. To solve this we use  Bayesian statistics-the posterior predictive distribution of possible unobserved values conditional on the observed values according to wikipedia. We will attempt to solve this problem using regression modelling and predictive mean matching using R software. Now linear regression has its limitations — it doesn’t allow for non-linear relationships and needs manual intervention to handle interactions .PMM is great  with non linear models. We will visualize missing data using various plots.

In this article we will cover these concepts extensively with each problems solving method.

#### Key takeaways 
-What is regression modelling
-The student will learn data cleaning methods
-The student will learn about some data visualizations in R
-The student will learn about predictive mean matching method to solve for missing data

#### Article quality
This article aims to make the learner able to solve missing data in datasets and each case method is well explained.

#### References
N/A

### PR DESCRIPTION
**Pull request by valentineomondi99**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6787

### CODE REVIEW

**Comment by valentineomondi99** [comment]
@louisefindlay23 , @mohamedgh16 I have renamed the file names using lowercase letters and also corrected the paragraph breaks. You can continue with the review .Thank you.

File: content/articles/predictive-mean-matching/index.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
+ In the real practical world, datasets have missing values/data which causes complications for data scientists who may deal with this by filling the datasets in an ad hoc fashion hence limiting them from their quest for data science main objectives; data visualization, data modeling, and giving conclusive meaningful summary or analysis of the given dataset.
+This tutorial aims at replacing the missing data using predictive mean matching and creating a predictive Regression model.
+We will explain the suitability, advantages, and limitation of each method for each use case.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:3]: Fix any punctuation errors in your introduction.

File: content/articles/predictive-mean-matching/index.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+### Goals
+By the end of this tutorial, the reader should be able to;
+- Understand and be able to carry out data imputation using these methods.
+- Handle missing data in datasets.
+- Visualize missing dataset using given plot functions.
+
+### Introduction
+In our tutorial, we will be using R programming language software to statistically impute missing values in datasets and carrying out an analysis of PMM and regression modeling and the suitability of these methods.
+Let's get started.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:37]: Move your introduction to the top. It should come as the first header (before the table of contents).

File: content/articles/predictive-mean-matching/index.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+### Installing and loading packages in R
+For our first step in this tutorial, we need to install and load packages that we will be using for our imputation and visualizing data.
+let's have the following packages installed;`VIM`,`mice`,`ggplot2`,we will use some functions in them.
+
+### Predictive mean matching and the use case
+It is an example of imputation that plugs the likely values of missing data by drawing randomly from a simulated observation whose values are closest to the missing data. We have singular imputation and multiple imputation.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:44]: Fix any spelling errors here and enhance the sentences.

File: content/articles/predictive-mean-matching/index.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+Writing the following code in R creates a predictive model with our interest modeled as:`chl` as the response variable and ` bmi`,`hyp` as the predicting variables.
+```r
+fitmodel <- with(imp, lm(chl  ~  bmi + hyp))
+summary(fitmodel)
+#we get the following below
+```
+![summary](/engineering-education/predictive-mean-matching/summary.PNG)
+
+The estimated effect of bmi on cholesterol level is 2.45, while the estimated effect of hyp on cholesterol is 16.845.
+Implying that for every 1% increase on bmi, there is a correlated 2.45% increase in the incidence of cholesterol levels.Also for every 1% increase in hyp, there is a 1% increase in the rate of cholesterol levels.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:99]: Fix the punctuation error here.

File: content/articles/predictive-mean-matching/index.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+### Multiple imputation by MICE
+MICE is a package that takes an incomplete dataset; having missing/NA values, and then plugs in the missing values through an appropriate technique.
+The default plugging technique is predictive mean matching with the `m` set to be equal to five as the default.
+For multiple imputation what it does is create a multiple number of complete datasets for each incomplete dataset then perfoms the required analysis to give a desired output.
+For this section we will call a dataset in R to show predictive mean matching by multiple imputation.
+For our tutorial, we choose the `enhans` dataset found in `mice` which has 4 variables `age`,`BMI`,`hypertension` and `cholesterol level`
+For our case, we will prioritize modeling cholesterol level as a function of age and hypertension.
+Let's load our data in R by following the steps.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:108]: This section has a lot of spelling errors, fix them.

File: content/articles/predictive-mean-matching/index.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+age  bmi  hyp  chl
+1  1  22.5  1  118
+2  2  22.7  1  187
+3  1  25.5  1  187
+4  3  27.4  1  218
+5  1  20.4  1  113
+6  3  21.7  1  184
+```
+We can go all the way up to the fifth instance.
+We can check for an imputed variable values stored in each of the 5 imputed datasets by running the following code in r.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:189]: Spelling error.

File: content/articles/predictive-mean-matching/index.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+11 187 199 187 206 131
+12 199 187 218 206 206
+15 187 186 187 199 187
+16 184 113 238 204 131
+20 186 218 184 186 199
+21 187 238 187 229 113
+24 184 204 187 206 284
+```
+For our next step, we append our imputed datasets with the original `nhanes` dataset.We carry out this using a function in `mice` called `complete()` function.we then use the `inc=TRUE` argument to specify what we want to combine with our original observed data.
+Let's run the following code in R to produce this result
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:206]: Missing "."

File: content/articles/predictive-mean-matching/index.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+```r
+imp2<- complete(imp, "long", inc = TRUE)
+#it is converted to a format that can be used in other statistical programs eg spss.
+#to view click `imp2` on the environment
+```
+
+### Fitting a linear regression model:A predictive model
+As we had stated earlier our aim is to build a predictive model with cholesterol levels as our response variable ,age and hypertension status as our predictor variables.Running the following code in R will approximate our regression model separately for the imputed datasets,from the first to the fifth.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:214]: Spelling errors.

**Review by ghost — COMMENTED** [review]
Great article @valentineomondi99 implement the corrections so we can move to tech review.

**Comment by valentineomondi99** [comment]
@mohamedgh16  I have made the changes as you requested. We can continue to the tech review. Thank you.

**Comment by valentineomondi99** [comment]
@mohamedgh16  ,any update on this PR ,I have implemented the requested changes waiting to go for the next step of tech tutorial 
cc: @louisefindlay23

**Comment by mohamedgh16** [comment]
Hello @valentineomondi99 please be patient as I'm reviewing other articles.

File: content/articles/predictive-mean-matching/index.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+10 (Intercept) 167. 21.8  7.66  0.000000120  25
+11  age  14.2  11.1  1.29  0.212  25
+12  hyp  3.78  21.1  0.179  0.859  25
+13 (Intercept) 127. 23.9  5.30  0.0000253  25
+14  age  32.9  10.9  3.01  0.00642  25
+15  hyp  2.58  19.8  0.130  0.898  25
+```
+
+### A complete case analysis for all five regressed models
+We write the following code in R to get this output.
```
Comment (mohamedgh16) [inline content/articles/predictive-mean-matching/index.md:237]: If you're pointing to an output use
 ```bash
 ```
 instead of 
 ```r

 ```

**Comment by valentineomondi99** [comment]
@mohamedgh16  I have made the requested changes you pointed out. we can continue thank you.

**Comment by valentineomondi99** [comment]
Hello @mohamedgh16  this article has been on the queue for quite sometime,kindly consider it soon for a final review.Thanks 
cc @louisefindlay23

**Comment by hectorkambow** [comment]
jumping on this one @section-engineering-education/content-moderator-team

**Comment by hectorkambow** [comment]
@valentineomondi99 

You may want to add a SKILL section to your author profile 🚀

**Pull request merged**